### PR TITLE
Update .gitignore

### PR DESCRIPTION
**Pull request by sleeepyjack**
Few minor updates to .gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,6 +49,7 @@ nosetests.xml
 coverage.xml
 *.cover
 .hypothesis/
+/tests/Testing/
```
Comment (sleeepyjack) [inline .gitignore:52]: This is somehow generated by `ctest`

File: .gitignore [hunk 2]
```diff
@@ -139,6 +140,14 @@ ENV/
 
 # clang
 compile_commands.json
+/.clangd/
```
Comment (sleeepyjack) [inline .gitignore:143]: Clangd language server config dir

File: .gitignore [hunk 3]
```diff
@@ -139,6 +140,14 @@ ENV/
...
 
 # figures
 *.eps
+
+# Github
+/.config/
```
Comment (sleeepyjack) [inline .gitignore:149]: Generated when using `gh` CLI client

File: .gitignore [hunk 4]
```diff
@@ -139,6 +140,14 @@ ENV/
...
+/.devcontainer.json
```
Comment (sleeepyjack) [inline .gitignore:150]: I run my own devcontainer config which lives at the project root dir.

File: .gitignore [hunk 5]
```diff
@@ -139,6 +140,14 @@ ENV/
...
+
+# AWS cache
+/.aws/
```
Comment (sleeepyjack) [inline .gitignore:153]: Coming from `sccache`

**Review by PointKernel — APPROVED** [review]
Ship it

**Pull request merged**